src/clinical/clinical-service.ts

```diff
@@ -263,7 +270,8 @@ export const getClinicalEntityMigrationErrors = async (
             readonly DeepReadonly<SchemaValidationError>[],
           ] = Object.entries(entityErrorObject)[0];
 
-          const entityName = currentEntityErrorData[0];
+          // todo: fix 'as'
+          const entityName = currentEntityErrorData[0] as ClinicalEntitySchemaNames;
```

Comment:
is it possible to make this fix or remove the todo? 
todos in code get lost to oblivion

Reply:
Yes let me give this a look today, Object.entries is what makes things challenging here

Reply:
OK this was a real challenge to figure out but I've definitely made improvements. If you can see how this is supposed to work better I'm open to changes.

A) Object.keys + Object.entries return `string` types for keys. This created a conflict where I tried to extract keys of type ClinicalEntitySchemaNames, because they would not be typed accurately downstream. A pattern around this is:

```
const { donorId, submitterDonorId, errors } = donor;
let entityName: ClinicalEntitySchemaNames;
errors.forEach((errorRecord) => {
          for (entityName in errorRecord) {
```
This is in src/clinical/clinical-service #266


B) TypeScript does not like dynamic variables, this example doesn't work for me, even though `schemaName`+`schemaNamesWithBreakingChanges` are typed correctly:

```
const donorSchemaErrors: DonorMigrationSchemaError[] = [];
for (const schemaName of schemaNamesWithBreakingChanges) {
      const errors = validateDonorEntityAgainstNewSchema(schemaName, newSchema, donor) || [];
      if (errors.length > 0) {
        donorSchemaErrors.push({
          [schemaName]: errors,
        });
      }
```

If we use `as` but only to create a new object it works fine:
```
for (const schemaName of schemaNamesWithBreakingChanges) {
      const errors = validateDonorEntityAgainstNewSchema(schemaName, newSchema, donor) || [];
      if (errors.length > 0) {
        const donorSchemaErrorRecord: DonorMigrationSchemaError = {} as DonorMigrationSchemaError;
        donorSchemaErrorRecord[schemaName] = errors;
        donorSchemaErrors.push(donorSchemaErrorRecord);
      }
```

Reply:
I also made updates to `DonorMigrationSchemaError`, trying to do cleanup related to changes in Jon's PR #980